[29] Cluster description sentences

georgerichardson opened the pull request:
---

Checklist:

- [x] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [ ] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [x] Appropriate information has been added to `README`s
- [ ] I have explained the feature in this PR or (better) in `output/reports/`
- [x] I have requested a code review

closes #29

georgerichardson committed:
Factor out clean scraped data getter

georgerichardson committed:
Add pipeline for generating scraped description sentence clusters

georgerichardson committed:
Add instructions for sentence clusters

georgerichardson committed:
Update to calculate token count per sentence

georgerichardson committed:
Fix typo

georgerichardson committed:
Add packages for sentence clusters

georgerichardson committed:
Add sentence cluster category distribution boxplots

georgerichardson committed:
Add getter for sentence clusters

georgerichardson committed:
Update with token counts

georgerichardson committed:
Add manually curated cluster categories

georgerichardson committed:
Fix formatting

georgerichardson committed:
Fix typo in cluster categories

Jack-0-0 commented on requirements.txt:
```diff
@@ -17,3 +17,6 @@ python-dotenv
 seaborn
 textstat
 tabulate
+umap-learn
+scikit-learn
+toolz
```
Need to add hdbscan to the requirements

Jack-0-0 reviewed (commented):
Thanks @georgerichardson.

The code looks good, there is just a missing requirement.

I tried to run the code. `comp_sci_gender_bias/pipeline/sentence_clusters/create_clusters.py` ran fine and produced the expected outputs.

When running `comp_sci_gender_bias/analysis/sentence_clusters.py` it has a KeyError for  `cluster_type` when running line 70 in function `category_token_distribution`.

Do you think it would be helpful to also add a readme on how you manually inspected the clusters and grouped them?

georgerichardson committed:
Notebook for reviewing and grouping sentence clusters

georgerichardson committed:
Add hdbscan

georgerichardson committed:
Add instructions for cluster grouping notebook

georgerichardson committed:
Merge branch 'dev' into 29_cluster_sents

georgerichardson commented:
> The code looks good, there is just a missing requirement.

Fixed, thanks.

> I tried to run the code. comp_sci_gender_bias/pipeline/sentence_clusters/create_clusters.py ran fine and produced the expected outputs.

👍 

> When running comp_sci_gender_bias/analysis/sentence_clusters.py it has a KeyError for cluster_type when running line 70 in function category_token_distribution.

I expect this is because you had just run `comp_sci_gender_bias/pipeline/sentence_clusters/create_clusters.py` which created new clusters. If there were fewer clusters than when I grouped them, then this would trigger a KeyError. You could test it by checking the output sentence data csv files back out from the branch. I don't like how fragile this is tbh.

> Do you think it would be helpful to also add a readme on how you manually inspected the clusters and grouped them?

I have added a small line to the readme which points to a notebook with the process that I followed.

Jack-0-0 reviewed (approved):
> > When running comp_sci_gender_bias/analysis/sentence_clusters.py it has a KeyError for cluster_type when running line 70 in function category_token_distribution.
> 
> I expect this is because you had just run `comp_sci_gender_bias/pipeline/sentence_clusters/create_clusters.py` which created new clusters. If there were fewer clusters than when I grouped them, then this would trigger a KeyError. You could test it by checking the output sentence data csv files back out from the branch. I don't like how fragile this is tbh.
> 

Ah, that makes sense. It runs fine with the original files in the outputs directory 🙂

> > Do you think it would be helpful to also add a readme on how you manually inspected the clusters and grouped them?
> 
> I have added a small line to the readme which points to a notebook with the process that I followed.

Great, thanks!

The PR looks good to merge.
